src/components/Records/Search/Input/FilterAutocomplete.vue

```diff
@@ -45,185 +45,187 @@
-             */
-            reset: function (selectedItem) {
-                selectedItem.filterSelected = {};
-            },
-            /**
-             * Clean up the given string by removing underscores and fills the stringsReplacement mapper variable.
-             * @param {String} string - the raw string to clean
-             * @returns {String} cleanedString - the string stripped of underscores.
-             */
-            cleanString: function (string) {
-                let cleanedString = string;
-                if (string.includes('_')) {
-                    cleanedString = string.replace(/_/g, " ");
-                }
-                return cleanedString;
+      if (Object.keys(currentParams).indexOf(filterName) === -1) {
+        if (_module.selectedValues !== null && _module.selectedValues.length > 0) {
+          if (_module.selectedValues.length === 1) {
+            currentParams[filterName] = encodeURIComponent(_module.selectedValues.join(','));
+          } else
+            {
+            let newParam = [];
+            _module.selectedValues.forEach(function (val) {
+              newParam.push(encodeURIComponent(val));
+            });
+            currentParams[filterName] = newParam.join(",");
+          }
+          _module.$router.push({
+            name: _module.$route.name,
+            query: currentParams
+          });
+        }
+      } else
+        {
+        if (_module.selectedValues === null || _module.selectedValues.length === 0)
+        {
+          delete currentParams[_module.filter.filterName];
+          _module.$router.push({
+            name: _module.$route.name,
+            query: currentParams
+          });
+        }
+        else
+        {
+          let newParams = [];
+          let existingValues = currentParams[_module.filter.filterName].split(",");
+          _module.selectedValues.forEach(function (selectedValue) {
+            const filterVal = encodeURIComponent(selectedValue);
+            if (existingValues.indexOf(filterVal) === -1) {
+              newParams.push(filterVal);
             }
+          });
+          currentParams[_module.filter.filterName] += `,${newParams.join(",")}`;
+          if (newParams.length > 0) {
+            _module.$router.push({
+              name: _module.$route.name,
+              query: currentParams
+            });
+          }
         }
+      }
+      _module.selectedValues = [];
+    },
+    /**
+     * Reset the form/filters/parameters to default (go so /search?page=1)
+     */
+    reset: function (selectedItem) {
+      selectedItem.filterSelected = {};
+    },
+    /**
+     * Clean up the given string by removing underscores and fills the stringsReplacement mapper variable.
+     * @param {String} string - the raw string to clean
+     * @returns {String} cleanedString - the string stripped of underscores.
+     */
+    cleanString: function (string) {
```

Comment:
Dont we have a mixin/utils for that now?

Reply:
done